pythonforandroid/bootstraps/service_only/build/src/org/kivy/android/PythonService.java

```diff
@@ -136,7 +131,6 @@ public void onDestroy() {
     @Override
     public void run() {
         PythonUtil.loadLibraries(getFilesDir());
-        mService = this;
```

Comment:
I understand your concern because of the change of behavior with regards to the old tool-chain.
The solution to this is a binder: https://developer.android.com/guide/components/bound-services.html

Example from http://stackoverflow.com/a/24541584
```
import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

public class MyService extends Service {
    private Binder binder;  

    @Override
    public void onCreate() {
        super.onCreate();
        binder = new Binder();
    }

    @Override
    public IBinder onBind(Intent intent) {
        return binder;
    }

    public class Binder extends android.os.Binder {
        public MyService getService() {
            return MyService.this;
        }
    }
}
```